MidasCivil_Adapter/CRUD/Create/Loads/BarVaryingDistributedLoads.cs

```diff
@@ -44,24 +49,44 @@ private bool CreateCollection(IEnumerable<BarVaryingDistributedLoad> barVaryingD
             foreach (BarVaryingDistributedLoad barVaryingDistributedLoad in barVaryingDistributedLoads)
             {
                 BarVaryingDistributedLoad load = barVaryingDistributedLoad.ShallowClone();
-                if (!load.RelativePositions)
+                
+                if (load.StartPosition >= load.EndPosition)
                 {
-                    Engine.Base.Compute.RecordError("The midas adapter can only handle BarVaryingDistributedLoads with relative positions. Please update the loads to be set with this format.");
-                    continue;
+                Engine.Base.Compute.RecordError("MidasCivil only supports start positions less than end positions for BarVaryingDistributedLoads.");
+                continue;
                 }
 
-                if (load.StartPosition >= load.EndPosition)
+                List<double> startPosition = new List<double>();
+                List<double> endPosition = new List<double>();
+
+
+                foreach (Bar bar in load.Objects.Elements)
                 {
-                    Engine.Base.Compute.RecordError("Midas civil only supports start positions less than end positions for BarVaryingDistributedLoads.");
-                    continue;
+                    if (load.RelativePositions == false)
+                    {
+                        double length = bar.Length();
+
+                        startPosition.Add(barVaryingDistributedLoad.StartPosition / length);
+                        endPosition.Add(barVaryingDistributedLoad.EndPosition / length);
+
+                        if (load.StartPosition / length > 1 || load.EndPosition / length > 1)
+                        {Engine.Base.Compute.RecordError("The load start or end position is outside one or more bars");}
```

Comment:
```suggestion
                        if (load.StartPosition / length > 1 || load.EndPosition / length > 1)
                        Engine.Base.Compute.RecordError("The load start or end position is outside one or more bars");
```
Should we not handle this before adding to this list - in my mind this is an error and should be dealt with.